### PR TITLE
Change MessageQueueMonitor from static to dynamic metric

### LINKED ISSUE #1683: Convert legacy static monitors to use dynamic metrics
### Describe the bug
There are some legacy code in metrics component that use static monitor to report interested metrics. As we know, those static metrics are hard to maintain and cause a lot of boilerplate code. We should convert them to use our dynamic metric framework so that our code base is clean and concise.

### To Reproduce
Check Helix code base for static monitors. 

### Expected behavior
After conversion, all metrics should work the same as before.

### PR DESCRIPTION
**Pull request by rabashizade**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:
Fixes #1683 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR modifies the MessageQueueMonitor class to use dynamic metrics by extending the DynamicMBeanProvider class. It also deprecates the MessageQueueMonitorMBean interface because it is no longer used.

### Tests

- [x] The following tests are written for this issue:

- The following is the result of the "mvn test" command on the appropriate module:

```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestExpandCluster.testClusterExpansion:61 expected:<false> but was:<true>
[ERROR]   TestDropCurrentStateRunningTask.testJobCurrentStateDroppedAfterCompletion:190 » Helix
[INFO] 
[ERROR] Tests run: 1265, Failures: 2, Errors: 0, Skipped: 2
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:47 h
[INFO] Finished at: 2021-04-26T17:44:23-07:00
[INFO] ------------------------------------------------------------------------
```

TestExpandCluster:
```
Shut down zookeeper at port 2183 in thread main
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 93.181 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
```
TestDropCurrentStateRunningTask:
```
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 80.98 s - in org.apache.helix.integration.task.TestDropCurrentStateRunningTask
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
```

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitorMBean.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 import org.apache.helix.monitoring.SensorNameProvider;
 
+@Deprecated
```
Comment (jiajunwang) [inline helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitorMBean.java:24]: Please remove it. There should be no direct reference to the MBean class. All usage should be done through the MBeanServer, so it is automatically backward compatible, and this class shall be removed.

**Review by zhangmeng916 — COMMENTED** [review]
LGTM. Could you please double check where the monitor gets used, including in the test, and make sure there won't be any regression.

File: helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitorMBean.java (hunk 1 above)
Comment (rabashizade) [reply helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitorMBean.java]: Thanks for the comment, I deleted the interface.

**Comment by rabashizade** [comment]
> LGTM. Could you please double check where the monitor gets used, including in the test, and make sure there won't be any regression.

Thanks for the comment. I double-checked, the monitor is only used in the HelixTaskExecutor class, and is only instantiated in 3 other places (MockHelixTaskExecutor, TestP2PNoDuplicatedMessage, and DefaultMessagingService) none of which resulted in any issues with changing the metric.

File: helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitor.java [hunk 2]
```diff
@@ -70,9 +72,9 @@ public void init() {
    * Remove this bean from the server
    */
   public void reset() {
-    _messageQueueBacklog = 0;
+    _messageQueueBacklog.updateValue(0L);
     try {
-      unregister(getObjectName(getBeanName()));
+      unregister();
     } catch (Exception e) {
       LOG.error("Fail to register MessageQueueMonitor", e);
```
Comment (dasahcc) [inline helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitor.java:79]: NIT: Am I missed something in another PR? I did not find the unregister() function.

Comment (rabashizade) [reply helix-core/src/main/java/org/apache/helix/monitoring/mbeans/MessageQueueMonitor.java:79]: It's defined in the parent class (DynamicMBeanProvider).

**Review by jiajunwang — APPROVED** [review]

**Comment by rabashizade** [comment]
This PR is ready to be merged, approved by @jiajunwang.

**Commit message:**

Change MessageQueueMonitor from static to dynamic metric 

This commit modifies the MessageQueueMonitor class to use dynamic metrics by extending the DynamicMBeanProvider class. Since after this change the MessageQueueMonitorMBean interface is no longer used in the code, this interface is deleted to remove all direct references to the MBean class.

**Pull request merged**